src/Service/SF1514/SF1514.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+/**
+ * This file is part of itk-dev/serviceplatformen.
+ *
+ * (c) 2020 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace ItkDev\Serviceplatformen\Service\SF1514;
+
+use DOMXPath;
+use ItkDev\Serviceplatformen\Certificate\CertificateLocatorInterface;
+use ItkDev\Serviceplatformen\Service\Exception\ServiceException;
+use ItkDev\Serviceplatformen\Service\SoapClient;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+class SF1514
+{
+    const TOKENTYPE_SAML20 = 'http://docs.oasis-open.org/wss/oasis-wss-saml-token-profile-1.1#SAMLV2.0';
+    const TOKENTYPE_STATUS = 'http://docs.oasis-open.org/ws-sx/ws-trust/200512/RSTR/Status';
+
+    const KEYTYPE_BEARER = 'http://docs.oasis-open.org/ws-sx/ws-trust/200512/Bearer';
+    const KEYTYPE_PUBLIC = 'http://docs.oasis-open.org/ws-sx/ws-trust/200512/PublicKey';
+
+    private SoapClient $soapClient;
+    private array $options;
+
+    public function __construct(SoapClient $client, array $options = [])
+    {
+        $this->soapClient = $client;
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+
+        $this->options = $resolver->resolve($options);
+    }
+
+    protected function getOption(string $name)
+    {
+        if (!array_key_exists($name, $this->options)) {
+            throw new ServiceException(sprintf('No such option: %s', $name));
+        }
+
+        return $this->options[$name];
+    }
+
+    public function isTestMode(): bool
+    {
+        return (bool)$this->options['test_mode'];
+    }
+
+
+    public function getSAMLToken(): ?string
```

Comment:
This function should throw an exception if we cannot get the SAML token.

Reply:
Fixed.